### ISSUE TITLE
How to reuse the function implemented by keras backend in evaluation ?

### ISSUE THREAD

**Issue by codingpy** [keras-team/keras#12458]
Because every time I use the Keras backend's function, I always can see, for example, `<tf.Tensor 'Const_3:0' ...>`, and the `Const_3` will be become `Const_4` when I called again!!
So, what I am worried about is that every time I call it, there is some logic record the operation or create some things. Maybe little by little, the more times I call the function in evaluation period, the slower the program it becomes.

Maybe you say that I can re-implement another function like this by using numpy instead,  but guess what happen if you use the same logic of the function both in training and evaluation period? It would be redundant to re-implement one in the same program.

Be more specifically, I use the following function respectively in calculating best `IOU` between ground truth and predictions during training, and in calculating `NMS` to filter some bad bounding boxes during evaluation.

If some good men would told me how to avoid redundant implementation? Or, maybe there is no cure but add a new functionality like `set_backend` that dynamically switch the backend from tensorflow to numpy's.

I would like to paste the function I use both in training and evaluation of Object Detection. There is the snippet:

```python
def IoU(box1, box2):
    xy1 = box1[..., :2]
    wh1 = box1[..., 2:]
    xy2 = box2[..., :2]
    wh2 = box2[..., 2:]

    xy3 = K.minimum(xy1+wh1/2, xy2+wh2/2)
    xy4 = K.maximum(xy1-wh1/2, xy2-wh2/2)
    wh3 = K.maximum(xy3-xy4, 0)

    a1 = K.prod(wh1, axis=-1)
    a2 = K.prod(wh2, axis=-1)
    a3 = K.prod(wh3, axis=-1)

    return a3 / (a1 + a2 - a3)

**Comment by xunnv** [comment]
Such a long time past, there are some ideas came to my mind that, should we have a NMS implemented in backend or other sub-module of Keras? Because, just using Keras backend (especially tensorflow's) can't do the job well.
There must exist a neat and gentle way to do NMS, and I hope some one could add this feature into Keras. Otherwise, everyone may implement themselves, or use CPU version of tensorflow's which is not flexible enough and doesn't allow user to specify IOU threshold dynamically.
